java/ql/src/semmle/code/java/dataflow/FlowSteps.qll

```diff
@@ -135,6 +135,15 @@ private class StringTaintPreservingConstructor extends Constructor, TaintPreserv
   override predicate returnsTaintFrom(int arg) { arg = 0 }
 }
 
+private class CharSequenceTaintPreservingMethod extends TaintPreservingCallable {
+  CharSequenceTaintPreservingMethod() {
+    this.getDeclaringType() instanceof TypeCharSequence and
```

Comment:
Should this consider subtypes as well (i.e. `getSourceDeclaration().getASourceSupertype*()`)?
It would then for example cover `CharBuffer`'s implementation of it, and would allow removing it from `StringTaintPreservingMethod`.

Reply:
yeah, good call, this is now part of the model